### PR TITLE
Add support for using updated_at without field value filter

### PR DESCRIPTION
**Pull request by jtruty**
Addresses issue noted here: https://github.com/fylein/sageintacct-sdk-py/issues/76

## Summary by CodeRabbit

- **New Features**
  - Improved filter logic in data fetching. Users can now more effectively combine `field`, `value`, and `updated_at` filters.

- **Chores**
  - Updated SDK version to 1.19.3.

### CODE REVIEW

**Review by ashwin1111 — APPROVED** [review]

**Comment by ashwin1111** [comment]
Thanks for the quick fix, @jtruty. I merged in a change today (which you're interested in - https://github.com/fylein/sageintacct-sdk-py/commit/6d5ad45269942273d6c28df69a0e1bb5f7d516a6) which caused some conflicts in your PR - could you please fix them up, we can merge and release it to PyPi.

**Comment by jtruty** [comment]
> Thanks for the quick fix, @jtruty. I merged in a change today (which you're interested in - [6d5ad45](https://github.com/fylein/sageintacct-sdk-py/commit/6d5ad45269942273d6c28df69a0e1bb5f7d516a6)) which caused some conflicts in your PR - could you please fix them up, we can merge and release it to PyPi.

Awesome, conflicts are fixed!

File: sageintacctsdk/apis/api_base.py [hunk 1]
```diff
@@ -468,35 +466,29 @@ def get_all_generator(self, field: str = None, value: str = None, fields: list =
...
+                    "and": {**field_value_filter, **updated_at_filter}
                 }
-                data['query']['filter']['and'].update(field_filter)
-                data['query']['filter']['and'].update(updated_at_filter)
-
-            if not data['query']['filter']:
-                del data['query']['filter']
-
-            if not data['query']['orderby']:
-                del data['query']['orderby']
```
Comment (ashwin1111) [inline sageintacctsdk/apis/api_base.py:499]: these blocks are required, else we might get error from APIs

```
{'message': 'Some of the parameters are wrong', 'response': {'error': [{'errorno': 'XL03000003', 'description': None, 'description2': "XML Parse schema error: Error 1871: Element 'orderby': Missing child element(s). Expected is ( order ).. Line: 1, column: 0. [Support ID: XRTTqWEB033~ZnpxnP5j0Br4ja4-a0TLJwAAABE]", 'correction': None}]}} {'error': [{'errorno': 'XL03000003', 'description': None, 'description2': "XML Parse schema error: Error 1871: Element 'orderby': Missing child element(s). Expected is ( order ).. Line: 1, column: 0. [Support ID: XYZ]", 'correction': None}]}

File: sageintacctsdk/apis/api_base.py [hunk 2]
```diff
@@ -468,35 +466,29 @@ def get_all_generator(self, field: str = None, value: str = None, fields: list =
...
+            # if we have updated_at_filter and field_value_filter we need to 'and' them
+            if updated_at_filter and field_value_filter:
+                data["query"]["filter"] = {
+                    "and": {**field_value_filter, **updated_at_filter}
                 }
-                data['query']['filter']['and'].update(field_filter)
-                data['query']['filter']['and'].update(updated_at_filter)
-
-            if not data['query']['filter']:
-                del data['query']['filter']
```
Comment (ashwin1111) [inline sageintacctsdk/apis/api_base.py:496]: same as below mentioned comment

**Review by ashwin1111 — COMMENTED** [review]
@jtruty please fix the couple of comments, we'll merge and release to PyPi once done

File: sageintacctsdk/apis/api_base.py (hunk 1 above)
Comment (jtruty) [reply sageintacctsdk/apis/api_base.py:499]: See here: https://github.com/fylein/sageintacct-sdk-py/pull/77/files#diff-c3a501ef4ad7ed46ca1eba1244d2d8797e11c44c0978a4d1ee2a8aadc37ca6ccL456-L459, we only set them if we need to, so a delete shouldn't be required.

Comment (ashwin1111) [reply sageintacctsdk/apis/api_base.py:499]: @jtruty we shouldn't send empty keys, the above error came up when I tried to send a request without order by value

Comment (jtruty) [reply sageintacctsdk/apis/api_base.py]: Got it - updated!

**Review by ashwin1111 — APPROVED** [review]

**Comment by ashwin1111** [comment]
@jtruty changes are released to PyPi - https://pypi.org/project/sageintacctsdk/

**Pull request merged**